src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/SimpleTaskQueue.cs

```diff
@@ -22,41 +22,25 @@ internal sealed class SimpleTaskQueue
         private readonly object _gate = new object();
 
         private Task _latestTask;
-        private int _taskCount;
 
         public SimpleTaskQueue(TaskScheduler taskScheduler)
         {
             _taskScheduler = taskScheduler;
-
-            _taskCount = 0;
             _latestTask = Task.CompletedTask;
         }
 
-        private TTask ScheduleTaskWorker<TArg, TTask>(Func<int, TArg, TTask> taskCreator, TArg arg)
-            where TTask : Task
-        {
-            lock (_gate)
-            {
-                _taskCount++;
-                var delay = (_taskCount % 100) == 0 ? 1 : 0;
```

Comment:
1. i agree this is bad.  but it also terrifies me to remove it.  
2. i have no idea why we need this SimpleTaskQueue.  It's really literally just two values.

Reply:
Ignore '2'.  it's used in enough places that it would be annoying to remove.  

My main take would be: This is a chesterton fence.  I'm really wary about touching those.  Especially in a type used in such deep places.

Reply:
this is code written by @CyrusNajmabadi I just renamed type/file to SimpleTaskQueue that was in workspace. I believe I fixed some bug around task, but logic and most of code is as it was written by Cyrus.

and it looks like he doesn't remember why it is there,

Reply:
> this is code written by @CyrusNajmabadi

I don't believe i wrote this... 

FWIW, this code predates even moving to github.  Do we have ways to even track past that point?

Reply:
Wait for my next PR :) I'm gonna nuke a lot of stuff here. Wanted this to be separate though.

Reply:
> i agree this is bad. but it also terrifies me to remove it.

I live on the edge ;-)